client.go

```diff
@@ -167,23 +178,31 @@ func (c *Client) Connect() error {
 		conf = &tls.Config{}
 	}
 	for {
-		conn, err := tls.DialWithDialer(dialer, "tcp", c.IrcAddress, conf)
-		c.connection = conn
+		if c.disconnected.get() {
+			return errors.New("client called Disconnect()")
```

Comment:
Since `Client::Connect` is a publicly callable, I think this error should be defined constantly in the package so it's easy for a user to check if it was the "client called Disconnect" error that was called without making a string compare